### PR TITLE
Minimum search terms

### PR DESCRIPTION
**Pull request by XavierJp**

### CODE REVIEW

File: cypress/e2e/advanced-research.cy.js [hunk 1]
```diff
@@ -29,3 +29,20 @@ paths.forEach((path) => {
     });
   });
 });
+
+describe('Minimum search conditions ' + path, () => {
+  it('No results if term < 3 and no filters', () => {
```
Comment (geoffreyaldebert) [inline cypress/e2e/advanced-research.cy.js:34]:
```suggestion
  it('No results if term length < 3 and no filters', () => {
```

File: cypress/e2e/advanced-research.cy.js [hunk 2]
```diff
@@ -29,3 +29,20 @@ paths.forEach((path) => {
...
+    cy.visit('/rechercher?terme=ga');
+    cy.contains('ne contient pas assez de paramètres').should('have.length', 1);
+  });
+
+  it('Results if term >= 3 and no filters', () => {
```
Comment (geoffreyaldebert) [inline cypress/e2e/advanced-research.cy.js:39]:
```suggestion
  it('Results if term length >= 3 and no filters', () => {
```

File: models/search.ts [hunk 3]
```diff
@@ -43,6 +45,13 @@ const search = async (
 ) => {
   const cleanedTerm = cleanSearchTerm(searchTerm);
 
+  const notEnoughParams =
+    searchTerm.length < 3 && !searchFilterParams.isNotEmpty();
+
+  if (notEnoughParams) {
+    return { ...noResults, notEnoughParams: true };
```
Comment (geoffreyaldebert) [inline models/search.ts:52]: A quoi ça sert ça ?

**Review by geoffreyaldebert — APPROVED** [review]

File: cypress/e2e/advanced-research.cy.js [hunk 4]
```diff
@@ -29,3 +29,20 @@ paths.forEach((path) => {
...
+    cy.visit('/rechercher?terme=aga');
+    cy.contains('ne contient pas assez de paramètres').should('have.length', 0);
+  });
+
+  it('Results if term < 3 and filters', () => {
```
Comment (geoffreyaldebert) [inline cypress/e2e/advanced-research.cy.js:44]:
```suggestion
  it('Results if term length < 3 and filters', () => {
```

**Comment by XavierJp** [comment]
Merging as test fail is related to firefox use. Will fix in different PR

**Pull request merged**